### ISSUE TITLE
Implement DisposeMethodShouldHaveBooleanParameterOverload

### ISSUE THREAD

**Issue by Vannevelj** [Vannevelj/VSDiagnostics#469]
> DO declare a protected virtual void Dispose(bool disposing) method to centralize all logic related to releasing unmanaged resources.

https://msdn.microsoft.com/en-us/library/b1yfkh5e(v=vs.110).aspx

**Comment by retailcoder** [comment]
Are you inspecting for usage of unmanaged resources? Isn't that virtual overload overkill otherwise?

**Comment by Vannevelj** [comment]
No, nothing about unmanaged inspection. You're right that it would be redundant but I think it's beneficial to have all IDisposable implementations be the same -- it prevents from a mistake creeping in there. And ultimately the dev can always ignore it of course.
